rllib/algorithms/dqn/dqn.py

```diff
@@ -722,9 +722,7 @@ def _training_step_new_api_stack(self):
                         for module_id, module_results in res.items():
                             if TD_ERROR_KEY in module_results:
                                 td_errors[module_id].extend(
-                                    convert_to_numpy(
-                                        module_results.pop(TD_ERROR_KEY).peek()
-                                    )
+                                    convert_to_numpy(module_results.pop(TD_ERROR_KEY))
```

Comment:
Today: In user-defined functions like training_step(), we assume that results a ResultsDict.
We break this assumption here because we call `MetricsLogger.reduce()` at the end of `Algorithm._run_one_training_iteration`.

With this PR, we instead return `MetricsLogger.compile()` in `Algorithm._run_one_training_iteration`.
In the future, we should probably standardize on using the MetricsLogger throughout everything under `Algorithm.step()` instead of ResultsDict and call MetricsLogger.compile at the end of `Algorithm.step()` to return a dict.